src/main/java/ch/saunah/saunahbackend/service/BookingService.java

```diff
@@ -47,66 +58,109 @@ public Booking addBooking(BookingBody bookingBody, int userId) throws Exception
+        booking.setComment(bookingBody.getComment());
+        booking.setGoogleEventID(calendarService.createEvent(sauna.getGoogleCalendarId(), booking));
+    }
 
-        return bookingRepository.save(booking);
+    private BookingPrice createBookingPrice(BookingBody bookingBody, Booking id) {
+        Price price = priceRepository.findAll().iterator().next();
+        if (price == null) {
+            throw new NotFoundException("No Price available in the database!");
+        }
+        BookingPrice bookingPrice = new BookingPrice();
+        bookingPrice.setTransportServicePrice(bookingBody.getTransportServiceDistance() * price.getTransportService());
+        bookingPrice.setWashServicePrice(price.getWashService());
+        bookingPrice.setSaunahImpPrice(bookingBody.getSaunahImpAmount() * price.getSaunahImp());
+        bookingPrice.setDepositPrice(price.getDeposit());
+        bookingPrice.setHandTowelPrice(bookingBody.getHandTowelAmount() * price.getHandTowel());
+        bookingPrice.setWoodPrice(bookingBody.getWoodAmount() * price.getWood());
+        bookingPrice.setDiscount(price.getDiscount());
+        return bookingPrice;
     }
 
-    private boolean dateRangeCollide(Date start, Date end, Date startExisting, Date endExisting) {
-        return start.getTime() >= startExisting.getTime() && start.getTime() <= endExisting.getTime() ||
-            end.getTime() >= startExisting.getTime() && end.getTime() <= endExisting.getTime() ||
-            start.getTime() < startExisting.getTime() && end.getTime() > endExisting.getTime();
+    private BookingSauna createBookingSauna(BookingBody bookingBody, Booking id) {
+        Sauna sauna = saunaService.getSauna(bookingBody.getSaunaId());
+        BookingSauna bookingSauna = new BookingSauna();
+        bookingSauna.setSaunaId(bookingBody.getSaunaId());
+        bookingSauna.setSaunaName(sauna.getName());
+        bookingSauna.setSaunaDescription(sauna.getDescription());
+        bookingSauna.setSaunaIsMobile(sauna.isMobile());
+        bookingSauna.setSaunaPrice(sauna.getPrice());
+        bookingSauna.setSaunaMaxTemp(sauna.getMaxTemp());
+        bookingSauna.setSaunaNumberOfPeople(sauna.getNumberOfPeople());
+        bookingSauna.setSaunaLocation(sauna.getLocation());
+        bookingSauna.setSaunaStreet(sauna.getStreet());
+        bookingSauna.setSaunaZip(sauna.getZip());
+        bookingSauna.setSaunaType(sauna.getType());
+        return bookingSauna;
+    }
+
+    /**
+     * Edit an already existing booking
+     *
+     * @param bookingId   the id of the booking to be edited
+     * @param bookingBody the parameter that shall be changed
+     * @return the booking that has been edited
+     */
+    public Booking editBooking(int bookingId, BookingBody bookingBody) throws IOException {
+        Booking editBooking = getBooking(bookingId);
+        setBookingFields(editBooking, bookingBody);
+        return bookingRepository.save(editBooking);
```

Comment:
validate bookingbody data, same as in addBooking -> create a validate method

Reply:
now a user can put in start dates before the current one which is not logical, and more...